### DISCUSSION TITLE
Double down on dynamic routing

### DISCUSSION THREAD

**Discussion by JudeTejada** [vercel/next.js#21962] [Help]
Quick question is it possible that we could double down on dynamic routing on nextjs?
where the role could be client/freelancer

/projects/:role/view/:id

I've been trying to set this up lately but I keep failing as there must be something that I'm doing wrong

initially what I had done was that I had a dynamic route on my project folder and on the view folder
But after accessing it on the route, the page couldn't able to be found

**Comment by stigkj** [comment]
You could let `view` be dynamic too, but return `notFound: true` if it is something other than `view`.

**Reply by stigkj** [reply]
@JudeTejada did you manage to get it working?

**Comment by michrome** [comment]
Hi @JudeTejada how about this?

```js
// pages/projects/[role]/view/[id].js

const View = ({ role, id }) => {
  switch (role) {
    case "client":
      return <p>This is CLIENT view of id {id}.</p>;
    case "freelancer":
      return <p>This is FREELANCER view of id {id}.</p>;
    default:
      return <p>You are not a client or a freelancer.</p>;
  }
};

export async function getServerSideProps({ query }) {
  return {
    props: { role: query["role"], id: query["id"] },
  };
}

export default View;
```

<img width="577" alt="Screenshot 2021-02-19 at 18 12 44" src="https://user-images.githubusercontent.com/596770/108544251-1ecd2880-72de-11eb-9690-4a56b41c817c.png">

and

```js
// pages/projects/[role]/index.js

const Role = ({ role }) => {
  switch (role) {
    case "client":
      return <p>This is CLIENT page.</p>;
    case "freelancer":
      return <p>This is FREELANCER page.</p>;
    default:
      return <p>You are not a client or a freelancer.</p>;
  }
};

export async function getServerSideProps({ query }) {
  return {
    props: { role: query["role"] },
  };
}

export default Role;
```

<img width="572" alt="Screenshot 2021-02-19 at 18 17 44" src="https://user-images.githubusercontent.com/596770/108544705-c4809780-72de-11eb-8c24-973cc0af1507.png">

**Reply by michrome** [reply]
Hi @JudeTejada did this answer your question?
